apps/website/src/components/shared/form/NumberField.tsx

```diff
@@ -47,34 +75,52 @@ export function NumberField(props: NumberFieldProps) {
 
   const { locale } = useLocale();
   const state = useNumberFieldState({ ...props, locale });
-  const { labelProps, inputProps } = useNumberField(
-    customizedProps,
-    state,
-    ref,
-  );
+  const {
+    labelProps,
+    groupProps,
+    inputProps,
+    incrementButtonProps,
+    decrementButtonProps,
+  } = useNumberField(customizedProps, state, ref);
+
+  const focusInput = useCallback(() => {
+    ref.current?.focus();
+  }, []);
 
   return (
     <div className={props.className || "flex-1"}>
-      <label {...labelProps}>{props.label}</label>
-      <div className="flex w-full items-center gap-1 rounded-sm border border-gray-700 bg-white text-gray-500">
-        {props.prefix}
-        <input
-          className={classes(
-            "w-full flex-1 bg-white p-1 px-2 text-black",
-            props.inputClassName,
+      <label className={props.labelClassName} {...labelProps}>
+        {props.label}
+      </label>
+      <div className={classes("flex", props.groupClassName)} {...groupProps}>
+        {props.showButtons && (
+          <NumberButton {...decrementButtonProps}>-</NumberButton>
+        )}
+        <div className="flex w-full items-center justify-center rounded-sm border border-gray-700 bg-white text-gray-500 focus-within:outline">
+          <div className="contents" onClick={focusInput}>
+            {props.prefix}
+            <input
+              className={classes(
+                "w-full flex-1 bg-white p-1 px-2 text-black focus:outline-0",
```

Comment:
I'm still seeing an outline here -- it seems to be `focus-within` that's causing it to show?

```suggestion
                "w-full flex-1 bg-white p-1 px-2 text-black outline-none",
```

Maybe this will ensure it never shows? I tried focus-withing:outline-0 but it doesn't seem to win.